org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/semantictokens/TokenType.java

```diff
@@ -32,7 +33,17 @@
 	PROPERTY("property"),
 	VARIABLE("variable"),
 	PARAMETER("parameter"),
-	MODIFIER("modifier");
+	MODIFIER("modifier"),
+
+	// Java-specific token types. These should be prefixed as such,
+	// to avoid confusion with similarly named concepts in other langugages.
+	// For example, "module" typically has the same meaning as "namespace"
+	// in other languages (Rust, for example). But in Java, modules are a
+	// completely separate concept, packages being the "namespace" tokens.
+	MODULE("javaModule"),
+	TYPE_LITERAL_KEYWORD("javaClassLiteralKeyword"), // genericName taken from: https://docs.oracle.com/javase/specs/jls/se11/html/jls-15.html#jls-15.8.2
```

Comment:
If I understand correctly, it's "class" part of "String.class". To highlight a "variable"-like "keyword", here:
* adds a new token type
* to avoid regression, clients are required to explicitly declare new types.
It makes not too much sense to me. Suggest to reuse "keyword" token.

Reply:
Yes, you've understood that correctly.
> to avoid regression, clients are required to explicitly declare new types. 
  It makes not too much sense to me. Suggest to reuse "keyword" token.

It's not a regression because the `class` postfix keyword (which technically isn't really a keyword, so if you have a better name please suggest one) was not previously part of semantic highlighting. In fact it wasn't even part of the VS Code Java TextMate grammar, which is the main reason I wanted to add it via semantic highlighting. The only reason I scope-map it in redhat-developer/vscode-java#1760 is so that existing themes will see it as a keyword, which does actually change the colorization, but for the better in my opinion (previously it was treated as a field/property, from the TextMate grammar, but that doesn't make sense to me). 

Using the `keyword` token is *fine*, but as I've said previously I don't know if I like the idea of making token types generalized when they can be specific. As far as the syntax highlighting in VS Code goes, I've already supertyped `javaClassLiteralKeyword` to `keyword` in redhat-developer/vscode-java#1760.